kubernetes/samples/scripts/common/utility.sh

```diff
@@ -185,6 +185,25 @@ function checkPvState {
   fi
 }
 
+#
+# Check the state of a persistent volume claim.
+# $1 - name of volume claim
+# $2 - expected state of volume claim
+function checkPvcState {
+
+  echo "Checking if the persistent volume claim ${1:?} is ${2:?}"
+  local pvc_state=`kubectl get pvc $1 -o jsonpath='{.status.phase}'`
+  attempts=0
+  while [ ! "$pvc_state" = "$2" ] && [ ! $attempts -eq 10 ]; do
+    attempts=$((attempts + 1))
+    sleep 1
+    pvc_state=`kubectl get pvc $1 -o jsonpath='{.status.phase}'`
+  done
```

Comment:
just a couple of minor comments: 

1) You can check "SECONDS" checks instead of using "attempt count" math to implement timeouts in shell, since SECONDS contains the time since the script started.  ("local end_secs=$(SECONDS + 10)" and "[ $SECONDS -le $end_secs]")

2) attempts should be declared local

Reply:
Cool, I love the `SECONDS` method, it will be more accurate.